classes/plugins/PluginHelper.php

```diff
@@ -72,17 +74,15 @@ private function extractPlugin(string $filePath, string $originalFileName): stri
             // Ensure there's a file named "version.xml" at the main directory or at the direct sub-directories
             foreach(new DirectoryIterator($extractPath) as $current) {
                 if ($current->isDir() && $current->getBasename() !== '..' && is_file(($path = "{$current->getPathname()}/") . static::PLUGIN_VERSION_FILE)) {
-                    return $path;
+                    return $onExtracted($path);
```

Comment:
The problem is exactly here, we retrieve the path where plugin was *found*, and the original extracted folder is kind of forgotten, it also happens at OJS 3.3.0, not a big deal as the main folder will be probably empty, but seeing this on the server made it a personal pet peeve 😁

Reply:
I'd really like to avoid further substantive changes for RC1 -- but as a caller of this code I wouldn't expect a callback. I think there are a few problems causing the complexity:
- The "create a container directory with a pseudorandom name" is wrapped into `extractPlugin`, rather than made available as a utility function and called outside of this function
- The "look for version.xml anywhere it might appear and return that directory" behaviour is unnecessary; we have a [standard toolset for building/releasing plugins](https://docs.pkp.sfu.ca/dev/plugin-guide/en/release#build-and-package-your-plugin) now and plugin installation code should expect plugins built accordingly.

Reply:
> (...) rather than made available as a utility function and called outside of this function

In general, I like such callbacks, especially in the form of language constructions (e.g. `using (var file = new File(...)) { code; }`), that disposes of the underlying resources, it decreases the amount of boilerplate code and brings extra safety (I'm unable to handle it wrong) 🤔

> The "look for version.xml anywhere it might appear and return that directory" behaviour is unnecessary

The previous code already had such checks in place, so I just made it more generic, I don't see it as a big deal, so I'm ok to keep these 3 lines :)

As the main ideas were preserved, it means this issue of leaving the main extracted folder forgotten, is also present at 3.3.0.